File: LeetCode/Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        for (int current_index = 0, last_zero_index = 0; current_index < nums.size(); current_index++) {
            if(nums[current_index] != 0) {
               swap(nums[last_zero_index++],nums[current_index]);
            }
        }
    }
};

File: LeetCode/Move_Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        curr_index = 0
        last_nonzero_index = 0
        for i in range(0,len(nums)):
            if nums[i] != 0:
                nums[i], nums[last_nonzero_index] = nums[last_nonzero_index], nums[i]
                last_nonzero_index +=1
